go/streamlog/streamlog_test.go

```diff
@@ -264,18 +265,39 @@ func TestFile(t *testing.T) {
 	}
 }
 
+func TestShouldSampleQuery(t *testing.T) {
+	queryLogSampleRate = -1
+	assert.False(t, shouldSampleQuery())
+
+	queryLogSampleRate = 0
+	assert.False(t, shouldSampleQuery())
+
+	// for test coverage, can't test a random result
```

Comment:
You can run 100 consecutive randoms and expect _some_ to pass and _some_ to fail. The false negative case for 1 in 2^100 is lower than any computational problem in modern hardware.